gen/templates/datasource.md.tmpl

```diff
@@ -46,24 +46,35 @@ Data source for ACI {{.ResourceNameAsDescription}}
 ```
 
 ## Schema ##
-
+{{ if or .HasNaming (and .HasParent (not (getDefaultValues .PkgName "parent_dn" $.Definitions)))}}
 ### Required ###
-{{ if .HasParent}}
+  {{ if and .HasParent (not (getDefaultValues .PkgName "parent_dn" $.Definitions))}}
```

Comment:
```suggestion
  {{- if and .HasParent (not (getDefaultValues .PkgName "parent_dn" $.Definitions))}}
```
Try the above change, which can reduce the unnecessary doc file changes.

Reply:
Fixed